PowerEditor/src/MISC/Common/Common.cpp

```diff
@@ -1331,23 +1331,37 @@ int nbDigitsFromNbLines(size_t nbLines)
 
 generic_string getDateTimeStrFrom(const generic_string& dateTimeFormat, const SYSTEMTIME& st)
 {
-	generic_string dateTimeStr = dateTimeFormat;
-	dateTimeStr = stringReplace(dateTimeStr, TEXT("Y"), std::to_wstring(st.wYear));
-	wchar_t buf[3];
-	_snwprintf(buf, sizeof(buf), TEXT("%02d"), st.wMonth);
-	dateTimeStr = stringReplace(dateTimeStr, TEXT("M"), buf);
-
-	_snwprintf(buf, sizeof(buf), TEXT("%02d"), st.wDay);
-	dateTimeStr = stringReplace(dateTimeStr, TEXT("D"), buf);
-
-	_snwprintf(buf, sizeof(buf), TEXT("%02d"), st.wHour);
-	dateTimeStr = stringReplace(dateTimeStr, TEXT("h"), buf);
-
-	_snwprintf(buf, sizeof(buf), TEXT("%02d"), st.wMinute);
-	dateTimeStr = stringReplace(dateTimeStr, TEXT("m"), buf);
+	const TCHAR* localeName = LOCALE_NAME_USER_DEFAULT;
+	const DWORD flags = 0;
+
+	constexpr int bufferSize = MAX_PATH;
+	TCHAR buffer[bufferSize] = {};
+	int ret = 0;
+
+	// 1. Remove format specifiers that we don't support:
+	//    - 'tt' that means AM/PM or 't' that means A/M
+	//    - 'g', 'gg' that means B.C. or A.D.
+	// Note: If 'tt' is supported it will need to be properly escaped
+	// to avoid conflict with 'M' date format that stands for month.
+	generic_string newFormat = dateTimeFormat;
+	auto it = std::remove_if(newFormat.begin(), newFormat.end(),
+		[](generic_string::value_type x) { return x == 't' || x == 'g'; });

```

Comment:
Removing AM & PM solve substitution of `M` problem. 
`g` can be restored without any consequence.